Nos/Views/DiscoverView.swift

```diff
@@ -290,35 +290,39 @@ struct DiscoverView_Previews: PreviewProvider {
         longNote.identifier = "2"
         longNote.author = user
         longNote.content = .loremIpsum(5)
-        
-        try! previewContext.save()
+
+        try? previewContext.save()
     }
     
     static func createRelayData(in context: NSManagedObjectContext, user: Author) {
         let addresses = ["wss://nostr.band", "wss://nos.social", "wss://a.long.domain.name.to.see.what.happens"]
         addresses.forEach {
-            _ = try! Relay(context: previewContext, address: $0, author: user)
+            _ = try? Relay(context: previewContext, address: $0, author: user)
         }
-        
-        try! previewContext.save()
+
+        try? previewContext.save()
     }
     
     @State static var relayFilter: Relay?
     
     static var previews: some View {
-        DiscoverView(featuredAuthors: [user.publicKey!.npub])
-            .environment(\.managedObjectContext, previewContext)
-            .environmentObject(relayService)
-            .environmentObject(router)
-            .environmentObject(currentUser)
-            .onAppear { createTestData(in: previewContext) }
-        
-        DiscoverView(featuredAuthors: [user.publicKey!.npub])
-            .environment(\.managedObjectContext, previewContext)
-            .environmentObject(relayService)
-            .environmentObject(router)
-            .environmentObject(currentUser)
-            .onAppear { createTestData(in: previewContext) }
-            .previewDevice("iPad Air (5th generation)")
+        if let publicKey = user.publicKey {
```

Comment:
Personally for SwiftUI previews I would rather deal just do force unwraps and get a crash and stack trace when something unexpected happens rather than trying to figure out why the canvas is empty. Not saying you should revert this but that's probably the route I'll take when writing preview code and unit tests.